### ISSUE TITLE
Better python documentation would be nice

### ISSUE THREAD

**Issue by Donald4444** [keyleds/keyleds#53]
I found out how to use the pythonbindings and think the documentation to change a single key is not quiet good. I will write here short how to change single keyleds. I hope this helps to improve the documentation or other people who couldn't figure it out.
`color = keyleds.KeyColor(42, 42, keyleds.Color(0, 255, 8))` (Found out with `dev.leds['keys'].get_all()`). `KEY_P` should be replaced with the id you got.
Then you need to make a list with the changes. In this case I only want one key to change, but you can add all changes to the list (I guess):
`list = [color]`
Then you add this to the changes that should be made:
`dev.leds['keys'].set_keys(list)`
and apply them:
`dev.commit_leds()`

**Comment by spectras** [comment]
Yes you're right, the python bindings were mostly intended as very low level thing, before the service was created.

Each method in the API corresponds to roughly one call to the USB hardware.
And as you noticed, the keyboard needs to be told to commit sent colors before it actually shows them, so this is also true from the python bindings.

A higher-level alternative would be to start the service and communicate with it over DBus. This allows many fancy things, such as writing animations in LUA and reacting to events.
